### DISCUSSION TITLE
End experiment if participant chooses not to consent

### DISCUSSION THREAD

**Discussion by hellenjingyuan** [jspsych/jsPsych#1004] [Q&A] [answered]
Hello! Sorry if this is a very basic question. At the beginning of my experiment, I present a consent form with two button-response options: "Yes, I consent to participate" and "No, I do not consent to participate". What I'm hoping to achieve is the following: if the participant clicks the consent button, the experiment proceeds. If the participant clicks the not-consent button, they are presented with some text ("You did not consent to participate. The session will now finish. Thank you for your time!") then the experiment ends. Any idea how I could achieve this? Thanks very much!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @hellenjingyuan ,

You can use the function `jsPsych.endExperiment(...)` inside the `on_finish` event of the button-response trial.

```js
on_finish: function(data){
	if(data.button_pressed == 1){
		jsPsych.endExperiment('<p>You did not consent to participate. The session will now finish. Thank you for your time!</p>');
	}
}
```

**Reply by hellenjingyuan** [reply]
Thank you very much @jodeleeuw! That works great.
